futures/tests/async_await_macros.rs

```diff
@@ -0,0 +1,48 @@
+#![feature(async_await, await_macro, pin, arbitrary_self_types, futures_api)]
+
+use futures::{Poll, pending, poll, pin_mut, join, select};
+use futures::channel::oneshot;
+use futures::executor::block_on;
+
+#[test]
+fn poll_and_pending() {
+    let pending_once = async { pending!() };
+    block_on(async {
+        pin_mut!(pending_once);
+        assert_eq!(Poll::Pending, poll!(&mut pending_once));
+        assert_eq!(Poll::Ready(()), poll!(&mut pending_once));
+    });
+}
+
+#[test]
+fn join() {
+    let (tx1, rx1) = oneshot::channel::<i32>();
+    let (tx2, rx2) = oneshot::channel::<i32>();
+
+    let fut = async {
+        let res = join!(rx1, rx2);
+        assert_eq!((Ok(1), Ok(2)), res);
+    };
+
+    block_on(async {
+        pin_mut!(fut);
+        assert_eq!(Poll::Pending, poll!(&mut fut));
+        tx1.send(1).unwrap();
+        assert_eq!(Poll::Pending, poll!(&mut fut));
+        tx2.send(2).unwrap();
+        assert_eq!(Poll::Ready(()), poll!(&mut fut));
+    });
+}
+
+#[test]
+fn select() {
+    let (tx1, mut rx1) = oneshot::channel::<i32>();
+    let (_tx2, mut rx2) = oneshot::channel::<i32>();
+    tx1.send(1).unwrap();
+    block_on(async {
+        select! {
+            rx1 => assert_eq!(Ok(1), rx1),
```

Comment:
It'd be good if the test could check that this arm is actually executed

Reply:
Done.